sound/soc/intel/boards/sof_cs42l42.c

```diff
@@ -641,6 +553,11 @@ static int sof_audio_probe(struct platform_device *pdev)
 			hdmi_num = 3;
 	}
 
+	if (mach->mach_params.common_hdmi_codec_drv)
```

Comment:
@kv2019i can you remind me which devices don't rely on the common hdac_hda code now? I wonder if we can start removing older stuff now.

Reply:
@plbossart I think we probably can. Especially for new mach drivers, I would drop hdac-hdmi support. The old driver was really only very briefly used with SOF by some Gemini Lake based Chromebooks (and non-SOF usage in SST), and transition mechanism was provided (we have no mach drivers for SOF that only work with the old hdac-hdmi). The change does impact the UCM files, so we proceeded very cautiously.

Reply:
I've got an GLK Chromebook and upgrade it's kernel to Chrome-v5.4 with SOF FW/TPLG based on stable-v2.2. The HDMI audio isn't using hdac-hdmi and is working good even with MST DP audio. All I need to do is removing mixers of hdac-hdmi from UCM file and changing the HDMI jack name. It seems to me that we could remove the hdac_hda support code from machine drivers.

Reply:
@brentlu you can keep the mixers and PCM but make them conditional in UCM, that way the UCM updates will work with all kernel versions.